docs/user-guide/v3_migration.rst

```diff
@@ -0,0 +1,204 @@
+     * ``zarr.hierarchy``
+     * ``zarr.indexing``
+     * ``zarr.meta``
+     * ``zarr.meta_v1``
+     * ``zarr.storage``
+     * ``zarr.sync``
+     * ``zarr.types``
+     * ``zarr.util``
+     * ``zarr.n5``
+
+3. Test that your package works with v3. You can start testing against version 3 now
+   (pre-releases are being published to PyPI weekly).
+4. Update the pin to zarr >=3
+
+Continue using Zarr-Python 2
+----------------------------
+
+Zarr-Python 2.x is still available, though we recommend migrating to Zarr-Python 3 for
+its improvements and new features. Security and bug fixes will be made to the 2.x series
+for at least 6 months following the first Zarr-Python 3 release.
+If you need to use the latest Zarr-Python 2 release, you can install it with:
+
+.. code-block:: console
+
+    $ pip install "zarr==2.*"
+
+Migration Guide
+---------------
+
+The following sections provide details on the most important changes in Zarr-Python 3.
+
+The Array class
+~~~~~~~~~~~~~~~
+
+1. Disallow direct construction - use :func:`zarr.open_array` or :func:`zarr.create_array`
+   instead of directly constructing the :class:`zarr.Array` class.
+
+2. Defaulting to ``zarr_format=3`` - newly created arrays will use the version 3 of the
+   Zarr specification. To continue using version 2, set ``zarr_format=2`` when creating arrays
+   or set ``default_zarr_version=2`` in :ref:`config`.
+
+The Group class
+~~~~~~~~~~~~~~~
+
+1. Disallow direct construction - use :func:`zarr.open_group` or :func:`zarr.create_group`
+   instead of directly constructing the :class:`zarr.Group` class.
+2. Deprecated most of the h5py compatibility methods. The following migration is suggested:
+
+   - Use :func:`zarr.Group.create_array` in place of :func:`zarr.Group.create_dataset`
+   - Use :func:`zarr.Group.require_array` in place of :func:`zarr.Group.require_dataset`
+
+The Store class
+~~~~~~~~~~~~~~~
+
+Some of the biggest changes in Zarr-Python 3 are found in the ``Store`` class. The most notable changes to the Store API are:
+
+1. Replaced the ``MutableMapping`` base class in favor of a custom abstract base class (:class:`zarr.abc.store.Store`).
+2. Switched to a primarily Async interface.
+
+Beyond the changes store interface, a number of deprecated stores were also removed in Zarr-Python 3:
+
+- ``N5Store``
+- ``DBMStore``
+- ``LMDBStore``
+- ``SQLiteStore``
+- ``MongoDBStore``
+- ``RedisStore``
+- ``ABSStore``
+
+Dependencies Changes
+~~~~~~~~~~~~~~~~~~~~
+
+- The new ``remote`` dependency group can be used to install a supported version of
+  ``fsspec``, required for remote data access.
+- The new ``gpu`` dependency group can be used to install a supported version of
+  ``cuda``, required for GPU functionality.
+- The ``jupyter`` optional dependency group has been removed, since v3 contains no
+  jupyter specific functionality.
+
+Configuration
+~~~~~~~~~~~~~
+
+There is a new configuration system based on `donfig <https://github.com/pytroll/donfig>`_,
```

Comment:
I think this should be in a what's new, not a migration guide, since there's no functionality from v2 in this section.